docs/developers/data/abi.md

```diff
@@ -279,3 +279,576 @@ Similarly, [enums](/developers/data/custom-types#custom-enums) are defined by:
+            "commitDate": "2023-05-25",
+            "channel": "Nightly",
+            "short": "rustc 1.71.0-nightly (a2b1646c5 2023-05-25)"
+        },
+        "contractCrate": {
+            "name": "adder",
+            "version": "0.0.0",
+            "gitVersion": "v0.44.0"
+        },
+        "framework": {
+            "name": "multiversx-sc",
+            "version": "0.44.0"
+        }
+    },
+    "docs": [
+        "One of the simplest smart contracts possible,",
+        "it holds a single variable in storage, which anyone can increment."
+    ],
+    "name": "Adder",
+    "constructor": {
+        "inputs": [
+            {
+                "name": "initial_value",
+                "type": "BigUint"
+            }
+        ],
+        "outputs": []
+    },
+    "endpoints": [
+        {
+            "name": "getSum",
+            "mutability": "readonly",
+            "inputs": [],
+            "outputs": [
+                {
+                    "type": "BigUint"
+                }
+            ]
+        },
+        {
+            "docs": [
+                "Add desired amount to the storage variable."
+            ],
+            "name": "add",
+            "mutability": "mutable",
+            "inputs": [
+                {
+                    "name": "value",
+                    "type": "BigUint"
+                }
+            ],
+            "outputs": []
+        }
+    ],
+    "events": [],
+    "esdtAttributes": [
+        {
+            "ticker": "testBasic",
+            "type": "BigUint"
+        }
+    ],
+    "hasCallback": false,
+    "types": {}
+}
+```
+We can also check the specific json file exported for the newly defined type where we can find information about the type separated from the main abi file.
+
+__testBasic.esdt-abi.json__
+```json
+{
+    "esdtAttribute": {
+        "ticker": "testBasic",
+        "type": "BigUint"
+    }
+}
+```
+
+## Composite types
+
+Now, let's see what happens when we use other types than basic ones. Let's add a `Vec`, an `Enum` and an `Option` to our esdt attributes.
+
+__adder.rs:__
+```rust
+#![no_std]
+
+use multiversx_sc::derive::TypeAbi;
+
+multiversx_sc::imports!();
+
+/// One of the simplest smart contracts possible,
+/// it holds a single variable in storage, which anyone can increment.
+#[multiversx_sc::contract]
+#[esdt_attribute("testBasic", BigUint)]
+#[esdt_attribute("testVec", ManagedVec<u64>)]
+#[esdt_attribute("testEnum", MyEnum)]
+#[esdt_attribute("testOption", Option<TokenIdentifier>)]
+pub trait Adder: module::ModuleExample {
+    #[view(getSum)]
+    #[storage_mapper("sum")]
+    fn sum(&self) -> SingleValueMapper<BigUint>;
+
+    #[init]
+    fn init(&self, initial_value: BigUint) {
+        self.sum().set(initial_value);
+    }
+
+    /// Add desired amount to the storage variable.
+    #[endpoint]
+    fn add(&self, value: BigUint) {
+        self.sum().update(|sum| *sum += value);
+    }
+}
+
+#[derive(TypeAbi)]
+pub enum MyEnum {
```

Comment:
We have an enum section above, `Custom types: enum`

You can add more examples there, but not here.